src/frontends/src/bench.cpp

```diff
@@ -3,21 +3,199 @@
+  std::cin >> num_threads;
+  std::cout << "Enter the number of request batches to be sent by each thread: ";
+  std::cin >> num_batches;
+  std::cout << "Enter the number of requests per batch: ";
+  std::cin >> batch_size;
+  std::cout << "Enter the delay between batches, in milliseconds: ";
+  std::cin >> batch_delay;
+
+  return create_config(path, num_threads, num_batches, batch_size, batch_delay);
+};
+
+std::unordered_map<std::string, std::string> get_config_from_json(std::string json_path) {
+  std::ifstream json_file(json_path);
+  std::stringstream buffer;
+  buffer << json_file.rdbuf();
+  std::string json_text = buffer.str();
+  rapidjson::Document d;
+  json::parse_json(json_text, d);
+  std::string cifar_path = json::get_string(d, "cifar_data_path");
+  std::string num_threads = json::get_string(d, "num_threads");
+  std::string num_batches = json::get_string(d, "num_batches");
+  std::string batch_size = json::get_string(d, "batch_size");
+  std::string batch_delay = json::get_string(d, "batch_delay_millis");
+  return create_config(cifar_path, num_threads, num_batches, batch_size, batch_delay);
+};
+
+int main(int argc, char *argv[]) {
+  cxxopts::Options options("performance_bench",
+                           "Clipper performance benchmarking");
+  // clang-format off
+  options.add_options()
+      ("f,filename", "Config file name", cxxopts::value<std::string>());
+  // clang-format on
+  options.parse(argc, argv);
+  bool json_specified = (options.count("filename") > 0);
+  std::unordered_map<std::string, std::string> test_config;
+
+  if(json_specified) {
+    std::string json_path = options["filename"].as<std::string>();
+    test_config = get_config_from_json(json_path);
+  } else {
+    test_config = get_config_from_prompt();
+  }
+  get_config().ready();
   QueryProcessor qp;
-  std::shared_ptr<Input> input =
-      std::make_shared<DoubleVector>(std::vector<double>{1.1, 2.2, 3.3, 4.4});
-  boost::future<Response> prediction =
-      qp.predict({"test", 3, input, 20000, "EXP3", {std::make_pair("m", 1)}});
-  log_info(LOGGING_TAG_CLIPPER, prediction.get().debug_string());
+  std::this_thread::sleep_for(std::chrono::seconds(3));
+  std::shared_ptr<metrics::RatioCounter> accuracy_ratio =
+      metrics::MetricsRegistry::get_metrics().create_ratio_counter("accuracy");
+  std::unordered_map<int, std::vector<std::vector<double>>> cifar_data = load_cifar(test_config);
+
+  int num_threads = std::stoi(test_config.find(CONFIG_KEY_NUM_THREADS)->second);
+  std::vector<std::thread> threads;
+  for (int i = 0; i < num_threads; i++) {
+    std::thread thread([&]() {
+      send_predictions(test_config, qp, cifar_data, accuracy_ratio);
+    });
+    threads.push_back(std::move(thread));
+  }
+  for (auto &thread : threads) {
+    thread.join();
+  }
+  std::string metrics = metrics::MetricsRegistry::get_metrics().report_metrics();
+  log_info("BENCH", metrics);
+  std::this_thread::sleep_for(std::chrono::seconds(5));
```

Comment:
I shouldn't be. I think this was an illogical attempt to log metrics in the face of the RPC error you identified above. Removed.